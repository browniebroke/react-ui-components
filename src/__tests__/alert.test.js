import React from 'react'
import { ThemeProvider } from 'styled-components'
import renderer from 'react-test-renderer'

import Alert from '../alert'
import styledComponentsTheme from '../styled-components-theme'

describe('Alert component', () => {
  const alertTypes = ['success', 'info', 'warning', 'danger']
  alertTypes.forEach((alertType) =>
    test(`should render correctly with ${alertType}`, () => {
      let tree = renderer
        .create(
          <ThemeProvider theme={styledComponentsTheme}>
            <Alert alertType={alertType}>This is a {alertType} alert</Alert>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  )

  const alignments = [null, '', 'left', 'center', 'right']
  alignments.forEach((textAlign) =>
    test(`should render correctly with ${textAlign}`, () => {
      let tree = renderer
        .create(
          <ThemeProvider theme={styledComponentsTheme}>
            <Alert alertType="info" textAlign={textAlign}>
              {textAlign} alert
            </Alert>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  )
})
