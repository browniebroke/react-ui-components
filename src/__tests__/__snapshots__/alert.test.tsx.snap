// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component should render correctly with  1`] = `
<div
  className="sc-bdfBwQ ieyMSl"
>
  
   alert
</div>
`;

exports[`Alert component should render correctly with center 1`] = `
<div
  className="sc-bdfBwQ ieyMSl"
>
  center
   alert
</div>
`;

exports[`Alert component should render correctly with danger 1`] = `
<div
  className="sc-bdfBwQ dXuoEv"
>
  This is a 
  danger
   alert
</div>
`;

exports[`Alert component should render correctly with info 1`] = `
<div
  className="sc-bdfBwQ ieyMSl"
>
  This is a 
  info
   alert
</div>
`;

exports[`Alert component should render correctly with left 1`] = `
<div
  className="sc-bdfBwQ fHBJUd"
>
  left
   alert
</div>
`;

exports[`Alert component should render correctly with right 1`] = `
<div
  className="sc-bdfBwQ exbWRu"
>
  right
   alert
</div>
`;

exports[`Alert component should render correctly with success 1`] = `
<div
  className="sc-bdfBwQ eZmQyz"
>
  This is a 
  success
   alert
</div>
`;

exports[`Alert component should render correctly with warning 1`] = `
<div
  className="sc-bdfBwQ cyqWze"
>
  This is a 
  warning
   alert
</div>
`;
